src/main/java/teammates/common/datatransfer/questions/FeedbackQuestionDetails.java

```diff
@@ -67,19 +67,26 @@ public abstract String getQuestionResultStatisticsCsv(List<FeedbackResponseAttri
     public abstract String getCsvHeader();
 
     /** Gets the header for detailed responses in csv format. Override in child classes if necessary. */
-    public String getCsvDetailedResponsesHeader(int noOfComments) {
-        return "Team" + "," + "Giver's Full Name" + ","
-               + "Giver's Last Name" + "," + "Giver's Email" + ","
-               + "Recipient's Team" + "," + "Recipient's Full Name" + ","
-               + "Recipient's Last Name" + "," + "Recipient's Email" + ","
-               + getCsvHeader()
-               + getCsvDetailedFeedbackResponsesCommentsHeader(noOfComments)
-               + System.lineSeparator();
+    public String getCsvDetailedResponsesHeader(int noOfInstructorComments, boolean isFeedbackParticipantCommentPresent) {
+        StringBuilder header = new StringBuilder(1000);
+        String headerString = "Team" + "," + "Giver's Full Name" + ","
+                + "Giver's Last Name" + "," + "Giver's Email" + ","
+                + "Recipient's Team" + "," + "Recipient's Full Name" + ","
+                + "Recipient's Last Name" + "," + "Recipient's Email" + ","
+                + getCsvHeader();
+        header.append(headerString);
+
+        if (isFeedbackParticipantCommentsOnResponsesAllowed() && isFeedbackParticipantCommentPresent) {
```

Comment:
For discussion.

So if there is no response comment for feedback participant, we will hide the comment header. Will this introduce unnecessary confusion? If I am an instructor, I'd rather see empty column (which means that nobody is explaining there answer) than having a question about whether the system has bug or not.

Reply:
Well, I was thinking what is the point of having empty column but your explanation seems convincing. Reverting this change.